src/Assessment/Assess.cs

```diff
@@ -566,6 +569,26 @@ public bool BeginAssessment()
             return true;
         }
 
+        private void UpdateExchangeRates()
+        {
+            UserInputObj.LoggerObj.LogInformation("Obtaining exchange rates' data");
+            double exchangeRate = 1.0;
+            try
+            {
+                exchangeRate = ForexData.GetExchangeRate(UserInputObj);
+            }
+            catch (OperationCanceledException)
+            {
+                throw;

```

Comment:
We should not throw. Maybe just log warning and keep using USD

Reply:
We only throw on "OperationCanceledException" -> This exception is caused when the user tries to "Stop" the process from the UI.

When the stop button is clicked, it sets a cancellationcontext. This cancellation context is read by all threads and periodically checked at a stable point during their execution. Once the cancellation context is set, OperationCancelled exception is thrown.